### PR TITLE
User and blog schema

### PR DESCRIPTION
**Pull request by blackjackkent**

### CODE REVIEW

**Comment by blackjackkent** [comment]
See #30 for previous discussion.

File: docker-compose.yml [hunk 1]
```diff
@@ -4,6 +4,9 @@ db:
     - "5432"
 app:
   build: .
+  command: bundle exec rails server -p 3000 -b '0.0.0.0'
```
Comment (0xazure) [inline docker-compose.yml]: Extraneous code from when we were setting up Docker stuff, no?

**Comment by 0xazure** [comment]
Regarding model tests: can we split the tests up by concern?  For example, we could have separate describe blocks for ActiveRecord associations and ActiveModel validations.  It would help keep like tests together, and future-proof against the time we have larger tests suites.

File: spec/models/blog_spec.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+require 'rails_helper'
+
+RSpec.describe Blog, type: :model do
+  describe 'initialization' do
```
Comment (0xazure) [inline spec/models/blog_spec.rb]: We don't need to nest everything under an initial `describe` block.

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  describe 'initialization' do
```
Comment (0xazure) [inline spec/models/user_spec.rb]: `describe` block nesting.

**Comment by 0xazure** [comment]
Last thing to nitpick may be RSpec test and block names.  Running rspec with the [documentation formatter](https://relishapp.com/rspec/rspec-core/v/2-5/docs/command-line/format-option#documentation-format) produces output like the following (note the test order is randomized):

```sh
$rspec --format=documentation

Randomized with seed 29976

User
  associations and object structure
    should have many blogs through blogs_users
    should have a secure password
  initialization
    has a valid factory
  validations
    should require email to be set
    is invalid without an password
    should require password to be set
    should require case sensitive unique value for email
    is invalid without an email

Blog
  validations
    is invalid without an username
    should require username to be set
    should require case sensitive unique value for username
  associations
    should have many users through blogs_users
  initialization
    has a valid factory
```

I think we should  endeavour to make these are readable as possible, since it's a nice way to get an overview of the thing under test and its expected behaviour and API interface.

**Comment by 0xazure** [comment]
Verified functionality :white_check_mark:  through manual test as [I detailed on #30](https://github.com/new-xkit/pipeline/pull/30#issuecomment-158268717), so this LGTM once we [finish nitpicking](https://github.com/new-xkit/pipeline/pull/37#issuecomment-163088344).

**Comment by blackjackkent** [comment]
I've updated the PR, per the above comments, with guidance from https://github.com/eliotsykes/rspec-rails-examples/blob/master/spec/models/subscription_spec.rb (mostly adding contexts and changing some describe phrasing, as well as switching more thoroughly to `expect` syntax (per http://betterspecs.org/#should ).

**Comment by 0xazure** [comment]
Updates to the spec LGM, ready to merge provided @blackjackkent you're happy with the commit history.

**Pull request merged**